crates/ibc/src/core/ics03_connection/connection.rs

```diff
@@ -502,9 +509,14 @@ impl State {
         }
     }
 
+    /// Returns true if this connection state matches the expected.
+    pub fn state_matches(&self, expected: &State) -> bool {
+        self == expected
+    }
```

Comment:
Just noticed this; this is a reimplementation of the standard `PartialEq::eq`. [3fa02e6](https://github.com/cosmos/ibc-rs/pull/626/commits/3fa02e6808ff51ca162f91b6d9fca90df2a7ca0d)